subgraphs/compound-forks/protocols/aurigami/src/mapping.ts

```diff
@@ -333,6 +350,184 @@ function getOrCreateProtocol(): LendingProtocol {
+    const rewards = getRewardsPerDay(
+      tryRewardSpeeds.value.plyRewardBorrowSpeed,
+      rewardToken
+    );
+    const mantissaFactorBD = exponentToBigDecimal(
+      18 - rewards.token.decimals + 18
+    );
+    const priceUSD = getPrice(PLY_MARKET, protocol._priceOracle)
+      .value.toBigDecimal()
+      .div(mantissaFactorBD);
+    rewardsAmount.push(rewards.rewardsPerDayBI);
+    rewardsAmountUSD.push(rewards.rewardsPerDayBD.times(priceUSD));
+    log.warning("price: {}, token: {}", [
+      priceUSD.toString(),
+      rewards.token.name,
+    ]);
+  }
+
+  if (tryRewardSpeeds.value.plyRewardSupplySpeed.gt(BIGINT_ONE)) {
+    const rewardToken = getOrCreateRewardToken(
+      PLY_TOKEN_ADDRESS,
+      RewardTokenType.DEPOSIT
+    );
+    rewardTokens.push(rewardToken.id);
+    const rewards = getRewardsPerDay(
+      tryRewardSpeeds.value.plyRewardSupplySpeed,
+      rewardToken
+    );
+    const mantissaFactorBD = exponentToBigDecimal(
+      18 - rewards.token.decimals + 18
+    );
+    const priceUSD = getPrice(PLY_MARKET, protocol._priceOracle)
+      .value.toBigDecimal()
+      .div(mantissaFactorBD);
+    rewardsAmount.push(rewards.rewardsPerDayBI);
+    rewardsAmountUSD.push(rewards.rewardsPerDayBD.times(priceUSD));
+    log.warning("price: {}, token: {}", [
+      priceUSD.toString(),
+      rewards.token.name,
+    ]);
+  }
+
+  if (tryRewardSpeeds.value.auroraRewardBorrowSpeed.gt(BIGINT_ONE)) {
+    const rewardToken = getOrCreateRewardToken(
+      AURORA_TOKEN_ADDRESS,
+      RewardTokenType.BORROW
+    );
+    rewardTokens.push(rewardToken.id);
+    const rewards = getRewardsPerDay(
+      tryRewardSpeeds.value.auroraRewardBorrowSpeed,
+      rewardToken
+    );
+    const mantissaFactorBD = exponentToBigDecimal(
+      18 - rewards.token.decimals + 18
+    );
+    const priceUSD = getPrice(AURORA_MARKET, protocol._priceOracle)
+      .value.toBigDecimal()
+      .div(mantissaFactorBD);
+    rewardsAmount.push(rewards.rewardsPerDayBI);
+    rewardsAmountUSD.push(rewards.rewardsPerDayBD.times(priceUSD));
+    log.warning("price: {}, token: {}", [
+      priceUSD.toString(),
+      rewards.token.name,
+    ]);
+  }
+
+  if (tryRewardSpeeds.value.auroraRewardSupplySpeed.gt(BIGINT_ONE)) {
+    const rewardToken = getOrCreateRewardToken(
+      AURORA_TOKEN_ADDRESS,
+      RewardTokenType.DEPOSIT
+    );
+    rewardTokens.push(rewardToken.id);
+    const rewards = getRewardsPerDay(
+      tryRewardSpeeds.value.auroraRewardSupplySpeed,
+      rewardToken
+    );
+    const mantissaFactorBD = exponentToBigDecimal(
+      18 - rewards.token.decimals + 18
+    );
+    const priceUSD = getPrice(AURORA_MARKET, protocol._priceOracle)
+      .value.toBigDecimal()
+      .div(mantissaFactorBD);
+    rewardsAmount.push(rewards.rewardsPerDayBI);
+    rewardsAmountUSD.push(rewards.rewardsPerDayBD.times(priceUSD));
+    log.warning("price: {}, token: {}", [
+      priceUSD.toString(),
+      rewards.token.name,
+    ]);
+  }
```

Comment:
If you want to reduce duplication you can do something like:

```
function calculateMarketRewards(token: Address, speed: BigInt, oracle: Address) {
  if (!speed.gt(BIGINT_ONE)) { return; }
  // get or create reward token
  // get rewards per day
  // get price use
  // rewardTokens.push(), rewardAmount.push() ….
  // etc
}
```

Reply:
Would I create a class to pass the arrays in and out of the function? THe assemblyscript kinda makes this tricky

Reply:
done